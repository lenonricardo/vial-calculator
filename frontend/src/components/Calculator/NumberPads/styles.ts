import styled from 'styled-components'
import Button from '@mui/material/Button'

export const Pad = styled(Button)`
  && {
    width: var(--pad-width);
    height: var(--pad-height);
    color: var(--font-pads);
    display: flex;
    font-size: var(--default-font-size);
    align-items: center;
    justify-content: center;
    cursor: pointer;
    user-select: none;
    border-radius: 0;

    &:hover {
      background-color: #fff;
    }
  }
`

export const PadWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const BottomPads = styled.div`
  display: flex;
  flex-direction: row;
  z-index: 2;
`

export const Pads = styled.div`
  transform: translateY(var(--pad-width));
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(var(--pad-width), 1fr));
  grid-template-rows: repeat(auto-fill, minmax(var(--pad-width), 1fr));
  width: calc(450px - var(--pad-width) - var(--pad-width));
  max-height: calc(450px - var(--pad-width));
`

export const DoublePad = styled(Pad)`
  && {
    width: calc(var(--pad-width) * 2);
  }
`
