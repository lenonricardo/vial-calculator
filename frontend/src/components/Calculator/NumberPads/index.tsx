import { PadWrapper, Pads, BottomPads, DoublePad, Pad } from './styles'

interface NumberPadsProps {
  onInput: (input: string) => void
}

enum NumberPadEnum {
  SEVEN = '7',
  EIGHT = '8',
  NINE = '9',
  FOUR = '4',
  FIVE = '5',
  SIX = '6',
  ONE = '1',
  TWO = '2',
  THREE = '3'
}

enum BottomPadEnum {
  ZERO = '0',
  DECIMAL_POINT = '.'
}

export default function NumberPads({ onInput }: NumberPadsProps) {
  const numberPads: NumberPadEnum[] = Object.values(NumberPadEnum)

  const handleNumberClick = (number: string) => {
    onInput(number)
  }

  return (
    <PadWrapper>
      <Pads>
        {numberPads.map((label) => (
          <Pad key={label} onClick={() => handleNumberClick(label)}>
            {label}
          </Pad>
        ))}
      </Pads>

      <BottomPads>
        <DoublePad onClick={() => handleNumberClick(BottomPadEnum.ZERO)}>
          {BottomPadEnum.ZERO}
        </DoublePad>
        <Pad onClick={() => handleNumberClick(BottomPadEnum.DECIMAL_POINT)}>
          {BottomPadEnum.DECIMAL_POINT}
        </Pad>
      </BottomPads>
    </PadWrapper>
  )
}
