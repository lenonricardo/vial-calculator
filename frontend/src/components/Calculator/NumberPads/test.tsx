import { render } from '@testing-library/react'

import NumberPads from '.'

describe('<NumberPads />', () => {
  it('should render the component', () => {
    const mockOnInput = jest.fn()

    const { container } = render(<NumberPads onInput={mockOnInput} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
