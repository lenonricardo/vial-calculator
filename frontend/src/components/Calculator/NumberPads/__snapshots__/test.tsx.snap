// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberPads /> should render the component 1`] = `
.c2 {
  width: var(--pad-width);
  height: var(--pad-height);
  color: var(--font-pads);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: var(--default-font-size);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  -webkit-transform: translateY(var(--pad-width));
  -ms-transform: translateY(var(--pad-width));
  transform: translateY(var(--pad-width));
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(var(--pad-width),1fr));
  grid-template-rows: repeat(auto-fill,minmax(var(--pad-width),1fr));
  width: calc(450px - var(--pad-width) - var(--pad-width));
  max-height: calc(450px - var(--pad-width));
}

.c4 {
  width: calc(var(--pad-width) * 2);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      7
    </div>
    <div
      class="c2"
    >
      8
    </div>
    <div
      class="c2"
    >
      9
    </div>
    <div
      class="c2"
    >
      4
    </div>
    <div
      class="c2"
    >
      5
    </div>
    <div
      class="c2"
    >
      6
    </div>
    <div
      class="c2"
    >
      1
    </div>
    <div
      class="c2"
    >
      2
    </div>
    <div
      class="c2"
    >
      3
    </div>
  </div>
  <div
    class="c3"
  >
    <div
      class="c2 c4"
    >
      0
    </div>
    <div
      class="c2"
    >
      .
    </div>
  </div>
</div>
`;
